packages/app/src/app/pages/Dashboard/Components/UpgradeBanner/UpgradeBanner.tsx

```diff
@@ -18,188 +18,104 @@ import { SUBSCRIPTION_DOCS_URLS } from 'app/constants';
-            },
-
-            [`@media screen and (min-width: ${WRAP_WIDTH}px)`]: {
-              gridColumnEnd: 'span 6',
-            },
-          }}
-          key={f.icon}
-          as="li"
-        >
-          <Stack css={{ color: '#999' }} gap={3}>
-            <Icon css={{ flexShrink: 0 }} name={f.icon} />
-            <Text size={3}>{f.label}</Text>
-          </Stack>
-        </Column>
-      ))}
-    </Grid>
-  );
-};
-
 export const UpgradeBanner: React.FC = () => {
-  const {
-    dashboard: { teams },
-  } = useAppState();
-  const { modalOpened, openCreateTeamModal } = useActions();
+  const history = useHistory();
   const [isBannerDismissed, dismissBanner] = useDismissible(
     'DASHBOARD_RECENT_UPGRADE'
   );
   const { isBillingManager, isPersonalSpace } = useWorkspaceAuthorization();
-  const { isEligibleForTrial } = useWorkspaceSubscription();
+  const {
+    isEligibleForTrial,
+    isInactiveTeam,
+    isLegacyFreeTeam,
+    isPro,
+  } = useWorkspaceSubscription();
+
+  const canStartTrial = isPersonalSpace || isEligibleForTrial;
   const { hasVisited } = useDashboardVisit();
 
   const [checkout, createCheckout, canCheckout] = useCreateCheckout();
 
-  if (isBannerDismissed || !hasVisited) {
+  if (isBannerDismissed || !hasVisited || isPro) {
     return null;
   }
 
-  const trialEligibleTeams = getTrialEligibleTeams(teams);
-
   const renderMainCTA = () => {
-    if (isPersonalSpace) {
+    // Dealing with workspaces that can upgrade to Pro
+    // Go directly to the stripe page for the checkout
+    if (canCheckout && (isInactiveTeam || isLegacyFreeTeam)) {
       return (
         <Button
+          disabled={checkout.status === 'loading'}
           css={{ padding: '4px 20px' }}
           onClick={() => {
-            if (trialEligibleTeams.length > 0) {
-              track('Home Banner - Start trial from personal workspace', {
+            if (isLegacyFreeTeam) {
+              const event = 'Home Banner - Start trial';
+              track(isBillingManager ? event : `${event} - As non-admin`, {
                 codesandbox: 'V1',
                 event_source: 'UI',
               });
-
-              modalOpened({ modal: 'selectWorkspaceToStartTrial' });
             } else {
-              track('Home Banner - Create team from personal workspace', {
+              track('Home Banner - Upgrade', {
                 codesandbox: 'V1',
                 event_source: 'UI',
               });
-
-              openCreateTeamModal();
             }
+
+            createCheckout({
+              utm_source: 'restrictions_banner',
+            });
           }}
           autoWidth
         >
-          Start free trial
+          {isEligibleForTrial ? 'Start 14-day free trial' : 'Upgrade'}
         </Button>
       );
     }
 
-    if (canCheckout) {
+    // Banner only shows for personal space, to upsell the pro workspaces
+    // Takes the user to the /pro page
+    // TODO: When can you start a free trial?
```

Comment:
Is this a pertinent todo?

Reply:
still discussing the requirement with @necoline